### PR TITLE
Allows installation of a defined PostgreSQL version from PGDG

### PR DESCRIPTION
**Pull request by lejmr**
This code is based on #70.

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+  RedHat:
+    - "postgresql{{ (postgresql_version|string).split('.')|join('') }}"
+    - "postgresql{{ (postgresql_version|string).split('.')|join('') }}-server"
+    - "postgresql{{ (postgresql_version|string).split('.')|join('') }}-contrib"
+    - "postgresql{{ (postgresql_version|string).split('.')|join('') }}-libs"
+  Debian:
+    - postgresql-{{ postgresql_version }}
+    - postgresql-contrib-{{ postgresql_version }}
+    - libpq-dev
+postgresql_oficial_repo_install: true
```
Comment (lhoss) [inline defaults/main.yml]: typo in `oficial`

File: defaults/main.yml [hunk 2]
```diff
@@ -1,6 +1,23 @@
 ---
 # RHEL/CentOS only. Set a repository to use for PostgreSQL installation.
 postgresql_enablerepo: ""
+postgresql_cluster_name: main
+postgresql_port: 5432
+
+# Postgresql.org repo path. Set a repository to use for PostgreSQL installation.
+postgresql_url_repo_path: "https://download.postgresql.org/pub/repos"
+postgresql_url_repo_key: "ACCC4CF8"
+postgresql_oficial_packages:
```
Comment (lhoss) [inline defaults/main.yml]: typo in `oficial`

File: templates/systemd.override.j2 [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[Service]
```
Comment (lhoss) [inline templates/systemd.override.j2]: Why is a custom systemd config required?  Do the official (pgdg) postgres /rh/centos) rpms not provide+install this?

File: vars/RedHat-7.yml [hunk 4]
```diff
@@ -1,9 +1,9 @@
 ---
 __postgresql_version: "9.2"
-__postgresql_data_dir: "/var/lib/pgsql/data"
+__postgresql_data_dir: "/var/lib/pgsql"
```
Comment (lhoss) [inline vars/RedHat-7.yml]: `/data` removed here (and not added later it seems) seems like a non downward-compatible change

**Comment by lejmr** [comment]
I've been out for vacation, so I didnt have time to wrap this up. Thank you for your feedback. Will try to incorporate your suggestions ASAP.

**Comment by lhoss** [comment]
this PR needs work: conflicts, requested fixes (like https://github.com/geerlingguy/ansible-role-postgresql/pull/73#discussion_r283412537)

**Comment by geerlingguy** [comment]
This introduces a huge architectural change to this role that I don't know if I'd want to support—basically a split of 'distro package' vs 'PGDG package'. IMO it is better to choose only one or the other for a role, otherwise this role will become unmaintainable.

It's already complex enough to try to account for the 12 (and counting) distro install methods... add on another 12 possibilities installing different versions from PGDG and it not only makes test runs take twice as long, it makes it mentally challenging to account for all the possibilities when making any future structural changes.

I would rather someone fork this role and use PGDG exclusively, that way this role can stay focused on postgres install from distro packaging, and that role can focus on install from PGDG.

Another option would be to make a few simple changes to make it possible for downstream users to switch to PGDG repos instead of distro repos using this same role structure, but even that could be a challenge.

**Comment by lejmr** [comment]
Fair enough. As long as my goal is to have a role that installs PostgreSQL from pgdg due to company's decision (we just want to have a fresh version of postgres while we believe to security updates from pgdg) .

I did not want to fork your role because I am not that much into PostgreSQL :-) Nevertheless, I think that is right way to go. Especially, if you do not mind that the role is going to contain about 90% of your code..

My biggest fear is how to support the further development. Right now your role has support for installation, but this is not always enough. Quite often I am using several extra options causing this role to break.. Therefore, If I actually start the pgdg fork I will start with tests.. then I believe we can merge these two roles because we will get the ground for merging these two roles together.. lets see if we are or not making a D tour right now :-)

**Comment by teadur** [comment]
I think that's quite common to run PGDG because they most likely will ship security patches faster then distro could.
I'm gonna try to merge the PR do my branch and see how it goes, but i doubt i have time to actually maintain a role for bigger audience.

**Pull request closed without merging**